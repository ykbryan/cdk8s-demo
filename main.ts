import { Construct } from 'constructs';
import { App, Chart, ChartProps } from 'cdk8s';
import { WebService } from './web-service';

export class MyChart extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = { }) {
    super(scope, id, props);

    // define resources here

    new WebService(this, 'sample-app', {
      image: 'bryanchua/sample-expressjs-microservice:latest',
      replicas: 2,
      containerPort: 3000
    })

    new WebService(this, 'sample-gateway', {
      image: 'bryanchua/sample-expressjs-gateway:latest',
      replicas: 2,
      containerPort: 3001
    })
    
  }
}

const app = new App();
new MyChart(app, 'cdk8s-demo');
app.synth();
