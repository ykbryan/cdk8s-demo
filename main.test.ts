import {MyChart} from './main';
import {Testing} from 'cdk8s';

describe('Placeholder', () => {
  test('Some Test', () => {
    const app = Testing.app();
    const chart = new MyChart(app, 'test-chart');
    const results = Testing.synth(chart)
    expect(results).toMatchSnapshot();
  });
});
