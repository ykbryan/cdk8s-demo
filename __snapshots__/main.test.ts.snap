// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder Some Test 1`] = `
Array [
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "name": "test-chart-sample-app-deployment-c82c4ed7",
    },
    "spec": Object {
      "replicas": 2,
      "selector": Object {
        "matchLabels": Object {
          "app": "test-chart-sample-app-c80c3b72",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "test-chart-sample-app-c80c3b72",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "image": "bryanchua/sample-expressjs-microservice:latest",
              "name": "sample-app",
              "ports": Array [
                Object {
                  "containerPort": 3000,
                },
              ],
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "name": "test-chart-sample-app-service-c89f2dd3",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "port": 80,
          "targetPort": 3000,
        },
      ],
      "selector": Object {
        "app": "test-chart-sample-app-c80c3b72",
      },
      "type": "LoadBalancer",
    },
  },
]
`;
